vc-got.el

```diff
@@ -636,7 +636,8 @@ Value is returned as floating point fractional number of days."
 
 (defun vc-got-delete-file (file)
   "Delete FILE locally and mark it deleted in work tree."
-  (vc-got--remove file t))
+  (let ((inhibit-read-only t))
+    (vc-got--remove file t)))
```

Comment:
I still haven't tried these functions, but the error you mention should be caused by `process-file` (in `vc-got--call`) trying to append `got rm` output in the `*vc-dir*` buffer.  You can probably avoid this and wrap `(vc-got--call "remove" ...)` in a `with-temp-buffer` (in `vc-got--remove`)

Reply:
ah, good catch, i added the with-temp-buffer and it seems to be working. I guess it could be later be extended to catch the output and return the got error if needed.

As far as delete-file this seems to be working.
unregister probably needs more work, I haven't found how / where it should be called.

Reply:
from a quick look in vc.el.gz it seems that unregister is only called in `vc-transfer-file`, to move a file between different vcs.  It isn't terribly useful